src/terminal/input/terminalInput.cpp

```diff
@@ -169,111 +185,143 @@ const TerminalInput::_TermKeyMap TerminalInput::s_rgModifierKeyMapping[]{
 {
     if (keyEvent.IsCursorKey())
     {
-        return (_fCursorApplicationMode) ? s_cCursorKeysApplicationMapping : s_cCursorKeysNormalMapping;
+        if (cursorApplicationMode)
+        {
+            return { s_cursorKeysApplicationMapping.data(), s_cursorKeysApplicationMapping.size() };
+        }
+        else
+        {
+            return { s_cursorKeysNormalMapping.data(), s_cursorKeysNormalMapping.size() };
+        }
+    }
+    else
+    {
+        if (keypadApplicationMode)
+        {
+            return { s_keypadApplicationMapping.data(), s_keypadApplicationMapping.size() };
+        }
+        else
+        {
+            return { s_keypadNumericMapping.data(), s_keypadNumericMapping.size() };
+        }
     }
-
-    return (_fKeypadApplicationMode) ? s_cKeypadApplicationMapping : s_cKeypadNumericMapping;
 }
 
-const TerminalInput::_TermKeyMap* TerminalInput::GetKeyMapping(const KeyEvent& keyEvent) const
+// Routine Description:
+// - Searches the keyMapping for a entry corresponding to this key event, and returns it.
+// Arguments:
+// - keyEvent - Key event to translate
+// - keyMapping - Array of key mappings to search
+// Return Value:
+// - Has value if there was a match to a key translation.
+std::optional<const TermKeyMap> _searchKeyMapping(const KeyEvent& keyEvent,
+                                                   std::basic_string_view<TermKeyMap> keyMapping) noexcept
 {
-    if (keyEvent.IsCursorKey())
+    bool keyTranslated = false;
+    for (auto& map : keyMapping)
     {
-        return (_fCursorApplicationMode) ? s_rgCursorKeysApplicationMapping : s_rgCursorKeysNormalMapping;
-    }
+        if (map.vkey == keyEvent.GetVirtualKeyCode())
+        {
+            // If the mapping has no modifiers set, then it doesn't really care
+            //      what the modifiers are on the key. The caller will likely do
+            //      something with them.
+            // However, if there are modifiers set, then we only want to match
+            //      if the key's modifiers are the same as the modifiers in the
+            //      mapping.
+            bool modifiersMatch = WI_AreAllFlagsClear(map.modifiers, MOD_PRESSED);
+            if (!modifiersMatch)
+            {
+                // The modifier mapping expects certain modifier keys to be
+                //      pressed. Check those as well.
+                modifiersMatch =
+                    (WI_IsFlagSet(map.modifiers, SHIFT_PRESSED) == keyEvent.IsShiftPressed()) &&
+                    (WI_IsAnyFlagSet(map.modifiers, ALT_PRESSED) == keyEvent.IsAltPressed()) &&
+                    (WI_IsAnyFlagSet(map.modifiers, CTRL_PRESSED) == keyEvent.IsCtrlPressed());
+            }
 
-    return (_fKeypadApplicationMode) ? s_rgKeypadApplicationMapping : s_rgKeypadNumericMapping;
+            if (modifiersMatch)
+            {
+                keyTranslated = true;
+                return map;

```

Comment:
`keyTranslated` is unused after this return, and it's .. also unused everywhere else.

Reply:
Woops.